pkg/crane/version.go

```diff
@@ -0,0 +1,36 @@
+package crane
+
+import (
+	"fmt"
+	"log"
+	"os/exec"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+// Version can be set via:
+// -ldflags="-X 'github.com/google/go-containerregistry/pkg/crane.Version=$TAG'"
+var Version string
+
+func init() { Root.AddCommand(NewCmdVersion()) }
+
+// NewCmdVersion creates a new cobra.Command for the version subcommand.
+func NewCmdVersion() *cobra.Command {
+	return &cobra.Command{
+		Use:   "version",
+		Short: "Print the version",
+		Args:  cobra.NoArgs,
+		Run: func(_ *cobra.Command, _ []string) {
+			if Version == "" {
+				// If Version is unset, use the current commit.
+				hash, err := exec.Command("git", "rev-parse", "HEAD").Output()
```

Comment:
As mentioned in https://github.com/google/ko/pull/25 (sorry I didn't notice it in this PR), we should try to run `git rev-parse HEAD` from within ko's directory, not within the current working directory, where it makes no sense.

Reply:
Oh wow I really wasn't thinking 🤦‍♂